### PR TITLE
Tslint

### PR DESCRIPTION
**Pull request by jonzee**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,7 +6,8 @@
   "scripts": {
     "build": "webpack --config webpack.config.js",
     "start": "node server.js",
-    "test": "mocha -r ts-node/register -r test-setup.js src/ts/**/*.spec.tsx"
+    "test": "npm run lint && mocha -r ts-node/register -r test-setup.js src/ts/**/*.spec.tsx",
```
Comment (michalmikolajczyk) [inline package.json]: we should most likely split this up into more tasks. The linter should check within the build process, not necessarily the tests, but still this is a step forward.

File: src/ts/API/index.tsx [hunk 2]
```diff
@@ -1,4 +1,4 @@
-let url = 'http://localhost:4000'
+const url = 'http://localhost:4000';
```
Comment (michalmikolajczyk) [inline src/ts/API/index.tsx]: ❤️

File: src/ts/AppBar/index.tsx [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
-import Menu from '../Menu'
-import Search from '../Search/Location'
-import muiThemeVG from '../muiThemeVG'
-import debug from 'debug'
-var log = debug('app:AppBar')
+import * as React from 'react';
+import { Link } from 'react-router';
+import { AppBar } from 'material-ui';
+import Menu from '../Menu';
+import { default as Search } from '../Search/Location';
```
Comment (michalmikolajczyk) [inline src/ts/AppBar/index.tsx]: whats that for?

File: src/ts/AppBar/index.tsx [hunk 4]
```diff
@@ -1,21 +1,20 @@
...
+import muiThemeVG from '../muiThemeVG';
+import debug from 'debug';
+const log = debug('app:AppBar');
 
 // Scroll logic based on https://gist.github.com/Warry/4254579
 export default class AppBarVG extends React.Component<any, any> {
-
-  static childContextTypes: any
+  static childContextTypes: any;
```
Comment (michalmikolajczyk) [inline src/ts/AppBar/index.tsx]: this line is not necessary

File: src/ts/AppBar/index.tsx [hunk 5]
```diff
@@ -1,21 +1,20 @@
...
+import { AppBar } from 'material-ui';
+import Menu from '../Menu';
+import { default as Search } from '../Search/Location';
+import muiThemeVG from '../muiThemeVG';
+import debug from 'debug';
+const log = debug('app:AppBar');
 
 // Scroll logic based on https://gist.github.com/Warry/4254579
 export default class AppBarVG extends React.Component<any, any> {
-
```
Comment (michalmikolajczyk) [inline src/ts/AppBar/index.tsx]: good! :)

File: src/ts/AppBar/index.tsx [hunk 6]
```diff
@@ -1,21 +1,20 @@
...
-  static childContextTypes: any
+  static childContextTypes: any;
 
   constructor(props) {
     super(props);
-    this.state = {visible: false};
-    this.handleScroll = this.handleScroll.bind(this)
+    this.state = { visible: false };
```
Comment (michalmikolajczyk) [inline src/ts/AppBar/index.tsx]: :heart:

File: src/ts/AppBar/index.tsx [hunk 7]
```diff
@@ -50,14 +49,14 @@ export default class AppBarVG extends React.Component<any, any> {
     }
   }
 
-  public search(filter) { log('AppBar search change', filter) }
+  public search(filter) { log('AppBar search change', filter); }
```
Comment (michalmikolajczyk) [inline src/ts/AppBar/index.tsx]: what's the `public` operator for?

File: src/ts/AppBar/index.tsx [hunk 8]
```diff
@@ -50,14 +49,14 @@ export default class AppBarVG extends React.Component<any, any> {
...
 
   render() {
-    const line = this.state.visible && <div className="app-bar-line"></div>
-    const search = this.state.visible && <Search icon="search" filter={this.search} />
+    const line = this.state.visible && <div className="app-bar-line"></div>;
+    const search = this.state.visible && <Search icon="search" filter={this.search} />;
     const icon = this.state.visible
```
Comment (michalmikolajczyk) [inline src/ts/AppBar/index.tsx]: if the only difference is in the className, perhaps we could just control the classname with this constant?

```
const icon = this.state.visible && ' border'
```
or even
```
<Link className={'title' + (this.state.visible && ' border')}" to="/">&#5809;</Link>
```

just make sure it does not render as the string literal `false`

File: src/ts/Login/Change.tsx [hunk 9]
```diff
@@ -1,40 +1,38 @@
...
+    super(props);
     this.state = {
       canSubmit: false,
       networkErr: false,
       changeError: false,
       changeSuccess: false,
-    }
-    this.submit = this.submit.bind(this)
+    };
+    this.submit = this.submit.bind(this);
```
Comment (michalmikolajczyk) [inline src/ts/Login/Change.tsx]: good good

File: src/ts/Login/Change.tsx [hunk 10]
```diff
@@ -1,40 +1,38 @@
...
-      } else {
-        this.setState({changeSuccess: true})
-      }
-    })
-    .catch(err => {
-      this.setState({networkErr: true})
-    })
+    const token = this.props.params.token;
+    change({ token, ...passwords })
+      .then((res) => {
```
Comment (michalmikolajczyk) [inline src/ts/Login/Change.tsx]: :heart:

File: src/ts/Login/Change.tsx [hunk 11]
```diff
@@ -1,40 +1,38 @@
...
+        if (res['err']) {
+          this.setState({ changeError: true });
+        } else {
+          this.setState({ changeSuccess: true });
+        }
+      })
+      .catch(err => this.setState({ networkErr: true }));
```
Comment (michalmikolajczyk) [inline src/ts/Login/Change.tsx]: or even
```
.catch(this.setState.bind(this, { networkErr: true }))
```
(if that would work? Or assign this bound function to a constant)

File: src/ts/AppBar/index.tsx (hunk 3 above)
Comment (jonzee) [inline src/ts/AppBar/index.tsx]: because of linter - it's allow us to import module only with the same name as in exported file (here is `Location`)

**Pull request merged**